File: Year II/First Semester/MAP/Laborator/Laborator9/ConsoleApp1/ConsoleApp1/Model/AbstractTaskContainer.cs
namespace ConsoleApp1.Model;

public abstract class AbstractTaskContainer: Container
{
    protected readonly List<Task> Tasks = [];

    public void Add(Task task)
    {
        Tasks.Add(task);
    }

    public bool Empty()
    {
        return Tasks.Count == 0;
    }

    public int Size()
    {
        return Tasks.Count;
    }

    public abstract Task Remove();
}

File: Year II/First Semester/MAP/Laborator/Laborator2/model/AbstractTaskContainer.java
package sem1_2.model;
import java.util.List;
import java.util.ArrayList;

public abstract class AbstractTaskContainer implements Container {
    List<Task> tasks = new ArrayList<>();

    @Override
    public abstract Task remove();

    @Override
    public void add(Task task) {
        tasks.add(task);
    }

    @Override
    public int size() {
        return tasks.size();
    }

    @Override
    public boolean isEmpty() {
        return tasks.isEmpty();
    }
}
